### DISCUSSION TITLE
Sonoff SV to control my not smart home vent keypad

### DISCUSSION THREAD

**Discussion by jat80** [arendst/Tasmota#21133] [Q&A]
Hi there,

I have a smart vent system that I'm wanting to control with a sonoff SV.  I've managed to find the keypad pins to turn the unit on and off and will model this with the sonoff sv but the issue is to turn it on I have to emulate a double key press (one to bring the screen on and a second to turn the power on). I then want the state of the switch to stay on until I send another two presses via the sonoff SV to turn it off again.

I've had a look at the blink and pulse time functions but unsure on how to keep the state On, or is there a way to configure the power on with two presses and off with two presses (bare in mind I'm not using the physical switch of the sonoff SV)

Sonoff SV is being used as a dry contact just need to connect the two wires on the smart vent board to toggle keypad 

Hopefully this makes sense and appreciate any feedback or help in the matter.

**Comment by barbudor** [comment]
1st in Tasmota vocabulary, a switch or a button in an Input, and an output is a Relay
So "I then want the state of the switch to stay on" doesn't mean anything

I understand you would like the POWER state to represent the actual state of your fan which is not directly possible as it represent the state of Tasmota's Relay output.

You could do something close by defining a 2nd relay output as a "virtual relay" on a GPIO that is unused and has nothing connected. Then you could change the state of this through rules to represent the state of your fan.
Still with rules you could manage to trigger the actual relay to send the pulses when this output is changed by either UI or MQTT.

Notes:
- Without getting a feedback in Tasmota of the actual state of the fan, you may get out of sync (ie showing ON when actually OFF,..)
- Your Sonoff SV will report to power state: the virtual fan state and the actual relay state. To be considered in case of any home system integration

**Comment by lalo-uy** [comment]
To prevent the out of sync I advice to get the fan power thru an optocupler
to a gpio input configured as switch. That way you have an input to know
its real state.
